pointpats/spacetime.py

```diff
@@ -629,3 +658,1008 @@ def _shuffle_matrix(X, ids):
+            "CartoDB Positron"
+        plot_edges : bool, optional
+            Whether to include lines drawn between members of a singnificant hotspot, by
+            default True
+        edge_weight : int, optional
+            line thickness when `plot_edges=True`, by default 2
+        edge_color : str, optional
+            color of line when `plot_edges=True`, by default "black"
+        colors : dict, optional
+            mapping of observation type to color,
+            by default {"focal": "red", "neighbor": "blue", "nonsig": "grey"}
+
+        Returns
+        -------
+        folium.Map
+            an interactive map showing locally-significant spacetime hotspots
+        """
+        if style_kwds is None:
+            style_kwds = {"radius": radius}
+        g = self._gdf.copy()
+
+        g["color"] = colors["nonsig"]
+
+        if inference == "permutation":
+            if not hasattr(self, "p_sims"):
+                warn(
+                    "Pseudo-p values not availalable. Permutation-based p-values require "
+                    "fitting the KnoxLocal class using `permutations` set to a large "
+                    "number. Using analytic p-values instead"
+                )
+                g["pvalue"] = self.p_hypergeom
+            else:
+                g["pvalue"] = self.p_sims
+        elif inference == "analytic":
+            g["pvalue"] = self.p_hypergeom
+        else:
+            raise ValueError("inference must be either `permutation` or `analytic`")
+
+        mask = g[g.pvalue <= crit].index.values
+        neighbors = self.adjlist[self.adjlist.focal.isin(mask)].neighbor.unique()
+
+        # this is clunky, but enforces plotting order so significance is prioritized
+        g.loc[neighbors, "color"] = colors["neighbor"]
+        g.loc[g.pvalue <= crit, "color"] = colors["focal"]
+        nbs = self.adjlist.groupby("focal").agg(list)["neighbor"]
+        g = g.reset_index().merge(nbs, left_on="index", right_index=True, how="left")
+
+        m = g[g.color == colors["nonsig"]].explore(
+            color="grey", style_kwds=style_kwds, tiles=tiles
+        )
+        blues = g[g.color == colors["neighbor"]]
+        if blues.shape[0] == 0:
+            warn("empty neighbor set.")
+        else:
+            m = blues.explore(m=m, color=colors["neighbor"], style_kwds=style_kwds)
+        m = g[g.color == colors["focal"]].explore(
+            m=m, color=colors["focal"], style_kwds=style_kwds
+        )
+
+        if plot_edges:
+
+            # edges between hotspot and st-neighbors
+            g = g.set_index("index")
+            ghs = self.hotspots(crit=crit, inference=inference)
+            ghs = ghs.dropna()
+            origins = g.loc[ghs.focal].geometry
+            destinations = g.loc[ghs.neighbor].geometry
+            ods = zip(origins, destinations)
+            lines = gpd.GeoSeries([LineString(od) for od in ods])
+            lines.crs = g.crs
```

Comment:
```suggestion
            lines = gpd.GeoSeries([LineString(od) for od in ods], crs=g.crs)
```

Do not use a direct assignment to `.crs` unless you have no other option (which you never have :)). It skips some internal checks in geopandas.

Reply:
ah, good to know. thanks